src/utils.ts

```diff
@@ -166,3 +166,27 @@ export async function genEnvFile(buildDir: string) {
     // Ignore errors, Meson could be too old to support --dump-format.
   }
 }
+
+export async function patchCompileCommands(buildDir: string) {
+  // Remove ccache from compile commands because they confuse Intellisense:
+  // https://github.com/microsoft/vscode-cpptools/issues/7616
+  try {
+    var filePath = path.join(buildDir, "compile_commands.json")
+    var db = JSON.parse(fs.readFileSync(filePath, "utf-8"));
+  } catch {
+    return;
+  }
+  var changed = false;
+  for (var i in db) {
+    // FIXME: This should use proper shlex.split() and shlex.join()
+    var cmd = db[i]['command'].split(" ");
```

Comment:
in these cases you should use `let` instead of `var`. `var` has some… interesting… behavior, which you're taking advantage of in the try/catch block

Reply:
var makes it available outside of the scope, like python, right?

Reply:
`var` does what's called "hoisting" basically anything declared with `var` is equivalent to being called with `let` at the very top of the function, uninitialized. so you can write something intresting like
```javascript
() => {
  console.log(foo);  // undefined, but exists
  var foo = 'foo';
}
```
because this is equivalent to
```javascript
() => {
  var foo;
  console.log(foo);
  foo = 'foo';
}
```

because of the hoisting, you also end up with var variables having function rather than block scope, so you can do such silliness as:
```javascript
const func = () => {
  let foo = () => { 
    var bar = 7;
    return bar
  };
  bar = 8;
  return foo;
}
func();  // returns 8
```